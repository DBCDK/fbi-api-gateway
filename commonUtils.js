import permissions from "./src/permissions.json";
export function parseClientPermissions({ smaug }) {
  return smaug?.gateway?.role
    ? permissions[smaug?.gateway?.role]
    : smaug?.gateway || permissions.default;
}
