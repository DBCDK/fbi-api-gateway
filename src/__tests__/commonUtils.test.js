import { parseClientPermissions } from "../../commonUtils";

describe("Permissions", () => {
  test("admin set to true in smaug configuration", async () => {
    const result = parseClientPermissions({
      smaug: { gateway: { admin: true } },
    });
    expect(result).toMatchSnapshot();
  });
  test("no gateway object in smaug configuration gives default permissions", async () => {
    const result = parseClientPermissions({
      smaug: {},
    });
    expect(result).toMatchSnapshot();
  });
  test("Empty gateway object in smaug configuration gives default permissions", async () => {
    const result = parseClientPermissions({
      smaug: { gateway: {} },
    });
    expect(result).toMatchSnapshot();
  });
  test("Role from gateway object in smaug configuration gives the roles' permissions", async () => {
    const result = parseClientPermissions({
      smaug: { gateway: { role: "ddbcms" } },
    });
    expect(result).toMatchSnapshot();
  });
  test("Unknown role from gateway object in smaug configuration gives default permissions", async () => {
    const result = parseClientPermissions({
      smaug: { gateway: { role: "unknown" } },
    });
    expect(result).toMatchSnapshot();
  });
  test("Explicitly setting allowRootFields in gateway object in smaug configuration gives those permissions", async () => {
    const result = parseClientPermissions({
      smaug: { gateway: { allowRootFields: ["manifestation"] } },
    });
    expect(result).toMatchSnapshot();
  });
});
