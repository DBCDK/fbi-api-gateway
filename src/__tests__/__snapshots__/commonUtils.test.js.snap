// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permissions Empty gateway object in smaug configuration gives default permissions 1`] = `
Object {
  "allowRootFields": Array [
    "debug",
    "draft",
    "manifestation",
    "manifestations",
    "search",
    "work",
    "works",
    "suggest",
    "complexSearch",
    "complexSearchIndexes",
    "complexSuggest",
    "localSuggest",
    "recommend",
    "recommendations",
    "infomedia",
    "refWorks",
    "ris",
    "elba",
    "library",
    "relatedSubjects",
    "linkCheck",
    "series",
    "universe",
    "mood",
  ],
  "denyTypes": Array [
    "CheckOrderPolicy",
    "Availability",
    "SEO",
    "Periodical",
    "PeriodicalArticle",
  ],
}
`;

exports[`Permissions Explicitly setting allowRootFields in gateway object in smaug configuration gives those permissions 1`] = `
Object {
  "allowRootFields": Array [
    "manifestation",
  ],
}
`;

exports[`Permissions Role from gateway object in smaug configuration gives the roles' permissions 1`] = `
Object {
  "allowRootFields": Array [
    "debug",
    "draft",
    "manifestation",
    "manifestations",
    "search",
    "work",
    "works",
    "suggest",
    "complexSearch",
    "complexSearchIndexes",
    "complexSuggest",
    "localSuggest",
    "recommend",
    "recommendations",
    "infomedia",
    "refWorks",
    "ris",
    "elba",
    "library",
    "relatedSubjects",
    "linkCheck",
    "series",
    "universe",
    "mood",
    "submitOrder",
  ],
  "denyTypes": Array [
    "CheckOrderPolicy",
    "Availability",
    "SEO",
    "Periodical",
    "PeriodicalArticle",
    "WorkReview",
  ],
}
`;

exports[`Permissions Unknown role from gateway object in smaug configuration gives default permissions 1`] = `
Object {
  "allowRootFields": Array [
    "debug",
    "draft",
    "manifestation",
    "manifestations",
    "search",
    "work",
    "works",
    "suggest",
    "complexSearch",
    "complexSearchIndexes",
    "complexSuggest",
    "localSuggest",
    "recommend",
    "recommendations",
    "infomedia",
    "refWorks",
    "ris",
    "elba",
    "library",
    "relatedSubjects",
    "linkCheck",
    "series",
    "universe",
    "mood",
  ],
  "denyTypes": Array [
    "CheckOrderPolicy",
    "Availability",
    "SEO",
    "Periodical",
    "PeriodicalArticle",
  ],
}
`;

exports[`Permissions admin set to true in smaug configuration 1`] = `
Object {
  "admin": true,
}
`;

exports[`Permissions no gateway object in smaug configuration gives default permissions 1`] = `
Object {
  "allowRootFields": Array [
    "debug",
    "draft",
    "manifestation",
    "manifestations",
    "search",
    "work",
    "works",
    "suggest",
    "complexSearch",
    "complexSearchIndexes",
    "complexSuggest",
    "localSuggest",
    "recommend",
    "recommendations",
    "infomedia",
    "refWorks",
    "ris",
    "elba",
    "library",
    "relatedSubjects",
    "linkCheck",
    "series",
    "universe",
    "mood",
  ],
  "denyTypes": Array [
    "CheckOrderPolicy",
    "Availability",
    "SEO",
    "Periodical",
    "PeriodicalArticle",
  ],
}
`;
